packages/api/cms-api/src/common/filter/enum.filter.factory.ts

```diff
@@ -21,17 +21,17 @@ export function createEnumFilter<TEnum extends { [key: string]: string }>(Enum:
             return "EnumFilter";
         }
 
-        @Field(() => [Enum], { nullable: true })
+        @Field(() => [Enum] as [{ [key: string]: string }], { nullable: true })
```

Comment:
What's the background behind this type cast? Looks kind of like a workaround

Reply:
It is a workaround. `@Field()` doesn't work correctly when using generics:

<img width="1103" alt="image" src="https://github.com/user-attachments/assets/6dd91032-8fbd-4912-879a-a9bc0da32e0d">

Alternatively, I could remove the generic from `createEnumFilter` altogether. I'm not sure if this is needed to be generic at all. What do you prefer?

Reply:
No it's okay. Let's keep it like that for now